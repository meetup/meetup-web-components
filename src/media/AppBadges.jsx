import PropTypes from 'prop-types';
import React from 'react';
import cx from 'classnames';

import Flex from '../layout/Flex';
import FlexItem from '../layout/FlexItem';

export const IOS_DOWNLOAD_LINK = 'https://meetu.ps/2ZhShs';
export const ANDROID_DOWNLOAD_LINK = 'https://meetu.ps/2ZhSyb';
const getAppStoreDownloadPhoto = (platform, language) =>
	// eslint-disable-next-line global-require, import/no-dynamic-require
	require(`../../assets/images/app_download/${platform}/download_${language}.png`);

/**
 * Renders iOS and Android app badges
 * @class AppBadges
 */
const AppBadges = ({
	className,
	language,
	getAppStorePhoto,
	isAndroidDevice,
	isIosDevice,
	isMobile,
	...other
}) => {
	const classNames = cx('getTheApp-downloadLinks', className);

	return (
		<Flex className={classNames} {...other}>
			{(isIosDevice || !isMobile) && (
				<FlexItem className="getTheApp-ios">
					<a href={IOS_DOWNLOAD_LINK} rel="noopener noreferrer" target="_blank">
						<img
							className="getTheApp-downloadImage span--100 margin--center"
							src={getAppStorePhoto('ios', language)}
						/>
					</a>
				</FlexItem>
			)}
			{(isAndroidDevice || !isMobile) && (
				<FlexItem className="getTheApp-android">
					<a
						href={ANDROID_DOWNLOAD_LINK}
						rel="noopener noreferrer"
						target="_blank"
					>
						<img
							className="getTheApp-downloadImage span--100 margin--center"
							src={getAppStorePhoto('android', language)}
						/>
					</a>
				</FlexItem>
			)}
		</Flex>
	);
};

AppBadges.defaultProps = {
	getAppStorePhoto: getAppStoreDownloadPhoto,
};

AppBadges.propTypes = {
	/** A function to load the app badge images - will likely take language and platform parameters */
	getAppStorePhoto: PropTypes.func,

	/** The language the badge image is in */
	language: PropTypes.string.isRequired,

	/** Checks if the user is using an iOS device*/
	isIosDevice: PropTypes.bool,

	/** Checks if the user is using an Android device*/
	isAndroidDevice: PropTypes.bool,

	/** Checks if the user is using a mobile device*/
	isMobile: PropTypes.bool,
};

export default AppBadges;
