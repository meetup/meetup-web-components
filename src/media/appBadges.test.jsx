import React from 'react';
import { shallow } from 'enzyme';

import AppBadges, {
	IOS_DOWNLOAD_LINK,
	ANDROID_DOWNLOAD_LINK,
} from './AppBadges';

const testLanguage = 'fr';

describe('App Badges', () => {
	const appBadgesComponent = shallow(<AppBadges language={testLanguage} />);
	it('exists', () => {
		expect(appBadgesComponent).toMatchSnapshot();
	});
	it('should link to the itunes store', () => {
		const itunesStoreLink = appBadgesComponent.find(
			`a[href="${IOS_DOWNLOAD_LINK}"]`
		);
		expect(itunesStoreLink.exists()).toBe(true);
	});
	it('should link to the google play store', () => {
		const googlePlayStoreLink = appBadgesComponent.find(
			`a[href="${ANDROID_DOWNLOAD_LINK}"]`
		);
		expect(googlePlayStoreLink.exists()).toBe(true);
	});
	it('should render the correct itunes and google play store images based on language', () => {
		const getAppStorePhoto = jest.fn();
		shallow(
			<AppBadges language={testLanguage} getAppStorePhoto={getAppStorePhoto} />
		);
		expect(getAppStorePhoto).toHaveBeenCalledWith('android', testLanguage);
		expect(getAppStorePhoto).toHaveBeenCalledWith('ios', testLanguage);
	});
});
