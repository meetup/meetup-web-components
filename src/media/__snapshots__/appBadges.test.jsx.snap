// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Badges exists 1`] = `
<Flex
  IsMobile={false}
  className="getTheApp-downloadLinks"
>
  <FlexItem
    className="getTheApp-ios"
  >
    <a
      href="https://meetu.ps/2ZhShs"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="Meetup iOS App Download"
        className="getTheApp-downloadImage span--100 margin--center"
        src=""
      />
    </a>
  </FlexItem>
  <FlexItem
    className="getTheApp-android"
  >
    <a
      href="https://meetu.ps/2ZhSyb"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="Meetup Android App Download"
        className="getTheApp-downloadImage span--100 margin--center"
        src=""
      />
    </a>
  </FlexItem>
</Flex>
`;
